src/main/java/seedu/address/model/person/Person.java

```diff
@@ -88,7 +94,8 @@ public boolean isSamePerson(Person otherPerson) {
         }
 
         return otherPerson != null
-                && otherPerson.getName().equals(getName());
+                && otherPerson.getName().equals(getName())
+                && otherPerson.getId() == getId();
```

Comment:
this method checks for duplicate persons in the address book when adding tutor or tutees, if you check for id which are different for every newly created person object, it won't return true

Reply:
Removed the getID() line